Automattic/pocket-casts-ios
# Playback 2024: Add multi-year support for End of Year/Playback

| 📘 Part of: #2250 |
|:---:|

Part of #2305

This refactors `EndOfYear` to include support for multiple years.

The intent is that this PR changes nothing with regards to the previous end of year projects. 2024 support will be in a separate PR due to line count.

## To test

> [!NOTE]  
> You will need to have sufficient listening history. Ping me if you want a database to import to speed that up.

* Enable `endOfYear` feature flag
* Tap the Reset modal/profile badge option in Developer menu
* Restart the app
* Navigate to Profile and ensure "Your Year in Podcasts" 2023 banner is showing
* Tap the banner and ensure Stories load and you can tap through them

## Checklist
